examples/testmonitor/results/Results.cs

```diff
@@ -0,0 +1,177 @@
+                        // Update the parent test step's status on the SystemLink server
+                        currentStep.Update(currentStepData);
+                    }
+                }
+
+                // If none of the child steps failed, mark the step as passed.
+                if (currentStepData.Status.StatusType.Equals(StatusType.Running))
+                {
+                    currentStepData.Status = new Status(StatusType.Passed);
+                    // Update the test step's status on the SystemLink server
+                    currentStep.Update(currentStepData);
+                }
+            }
+
+            // Update the top-level test result's status based on the most severe child step's status
+            testResult = testResult.DetermineStatusFromSteps();
+        }
+
+        /// <summary>
+        /// Simulates taking an electrical power measurement.
+        /// This introduces some random current and voltage loss.
+        /// </summary>
+        /// <param name="current">The electrical current value.</param>
+        /// <param name="voltage">The electrical voltage value.</param>
+        /// <returns>The a tuple containing the electrical power measurements and the input and output lists.</returns>
+        private static (double power, List<NamedValue> inputs, List<NamedValue> outputs) MeasurePower(int current, int voltage = 0)
+        {
+            var random = new Random();
+            var currentLoss = 1 - random.NextDouble() * 0.25;
+            var voltageLoss = 1 - random.NextDouble() * 0.25;
+            var power = current * currentLoss * voltage * voltageLoss;
+
+            // Record electrical current and voltage as inputs
+            var inputs = new List<NamedValue>()
+            {
+                new NamedValue("current", current),
+                new NamedValue("voltage", voltage)
+            };
+
+            // Record electrical power as an output
+            var outputs = new List<NamedValue>()
+            {
+                new NamedValue("power", power)
+            };
+
+            return (power, inputs, outputs);
+        }
+
+        /// <summary>
+        /// Builds a Test Monitor measurement parameter object for the power test.
+        /// </summary>
+        /// <param name="power">The electrical power measurement.</param>
+        /// <param name="lowLimit">The value of the low limit for the test.</param>
+        /// <param name="highLimit">The value of the high limit for the test.</param>
+        /// <param name="status">The measurement's pass/fail status.</param>
+        /// <returns>A list of test measurement parameters.</returns>
+        private static List<Dictionary<string, string>> BuildPowerMeasurementParams(double power, double lowLimit, double highLimit, Status status)
+        {
+            var parameter = new Dictionary<string, string>();
+            parameter.Add("name", $"Power Test");
```

Comment:
Not super necessary for this example, but just FYI - there's a neat initializer syntax for this:

```
            var parameter = new Dictionary<string, string>
            {
                ["name"] = "Power Test",
                ...
            };
```

Reply:
Nice, I like it.  I'll push and update for it.